_pytest/assertion/rewrite.py

```diff
@@ -151,6 +136,32 @@ def find_module(self, name, path=None):
         self.modules[name] = co, pyc
         return self
 
+    def _should_rewrite(self, fn_pypath, state):
+        # always rewrite conftest files
+        fn = str(fn_pypath)
+        if fn_pypath.basename == 'conftest.py':
+            state.trace("rewriting conftest file: %r" % (fn,))
+            return True
+        elif self.session is not None:
+            if self.session.isinitpath(fn):
+                state.trace("matched test file (was specified on cmdline): %r" %
+                            (fn,))
+                return True
+            else:
+                # modules not passed explicitly on the command line are only
+                # rewritten if they match the naming convention for test files
+                session = self.session  # avoid a cycle here
+                self.session = None
+                try:
+                    for pat in self.fnpats:
+                        if fn_pypath.fnmatch(pat):
+                            state.trace("matched test file %r" % (fn,))
+                            return True
+                finally:
+                    self.session = session
+                    del session
```

Comment:
This seems superfluous as the function returns anyways, so `session` gets out of scope